# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Parser.cpp
# Commit: b4f304b80390
# Full Hash: b4f304b803904a9d73cd9c69b0ed66f78db5c039
# Author: Jonatan Klemets <jonatan.r.klemets@gmail.com>
# Date: 2022-04-14 21:45:12
# Description:
#   Bug 1764737 - Add missing OOM checks in GeneralParser<ParseHandler, Unit>::exportFrom. r=mgaudet
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D143751
# ==============================================================================

diff -r ed4101ff3fae -r b4f304b80390 js/src/frontend/Parser.cpp
--- a/js/src/frontend/Parser.cpp	Thu Apr 14 19:02:06 2022 +0000
+++ b/js/src/frontend/Parser.cpp	Thu Apr 14 19:08:03 2022 +0000
@@ -5703,6 +5703,9 @@
 
   ListNodeType importAssertionList =
       handler_.newList(ParseNodeKind::ImportAssertionList, pos());
+  if (!importAssertionList) {
+    return null();
+  }
   if (tt == TokenKind::Assert) {
     tokenStream.consumeKnownToken(TokenKind::Assert,
                                   TokenStream::SlashIsRegExp);
@@ -5718,6 +5721,9 @@
 
   BinaryNodeType moduleRequest = handler_.newModuleRequest(
       moduleSpec, importAssertionList, TokenPos(moduleSpecPos, pos().end));
+  if (!moduleRequest) {
+    return null();
+  }
 
   BinaryNodeType node =
       handler_.newExportFromDeclaration(begin, specList, moduleRequest);